internal/providercache/package_install.go

```diff
@@ -37,12 +99,17 @@ func installFromHTTPURL(ctx context.Context, meta getproviders.PackageMeta, targ
 	// through X-Terraform-Get header, attempting partial fetches for
 	// files that already exist, etc.)
 
-	httpClient := httpclient.New()
+	retryableClient := retryablehttp.NewClient()
+	retryableClient.HTTPClient = httpclient.New()
+	retryableClient.RetryMax = maxRetryCount
```

Comment:
I wonder if some of these should not be moved up to the PackageMeta, but I'm not very well versed in this part of the code. @cam72cam you know this better, could you take a look please?

Reply:
@janosdebugs @cam72cam currently i am following the pattern that we used in codebase, Yes i agree there are some code around retry mechanism that can be refactor.

Reply:
This looks fine to me!